onnxruntime/python/onnxruntime_pybind_state.cc

```diff
@@ -1601,11 +1602,12 @@ including arg name, arg type (contains both type and shape).)pbdoc")
           },
           py::return_value_policy::reference_internal)
       .def_property_readonly(
-          "session_options", [](const PyInferenceSession* sess) -> const PySessionOptions& {
-            const auto& session_options = sess->GetSessionHandle()->GetSessionOptions();
-            return static_cast<const PySessionOptions&>(session_options);
+          "session_options", [](const PyInferenceSession* sess) -> const PySessionOptions* {
```

Comment:
>const [](http://example.com/codeflow?start=67&length=5)

Suggest not to return const for readability although with pybind it is all inconsequential.

Reply:
Updated to non-const return type